src/auth/connection.ts

```diff
@@ -24,10 +24,33 @@ export const ssoAccountAccessScopes = ['sso:account:access']
 export const codewhispererScopes = ['codewhisperer:completions', 'codewhisperer:analysis']
 export const defaultSsoRegion = 'us-east-1'
 
+type SsoType = 'any' | 'base' | 'builderId'
+
 export const isIamConnection = (conn?: Connection): conn is IamConnection => conn?.type === 'iam'
-export const isSsoConnection = (conn?: Connection): conn is SsoConnection => conn?.type === 'sso'
-export const isBuilderIdConnection = (conn?: Connection): conn is SsoConnection =>
-    isSsoConnection(conn) && conn.startUrl === builderIdStartUrl
+export const isSsoConnection = (conn?: Connection, type: SsoType = 'any'): conn is SsoConnection => {
+    if (conn?.type !== 'sso') {
+        return false
+    }
+    // At this point the conn is an SSO conn, but now we must determine the specific type
+    switch (type) {
+        case 'base':
+            // Any SSO type is considered an SSO connection, but in this case
+            // we only want an Identity Center connection. So we return true
+            // as long as the connection is none of the other connection types.
+
+            // IMPORTANT: This condition should grow as more SsoType's get added.
+            return !isSsoConnection(conn, 'builderId')
```

Comment:
nit: does recursion reduce the clarity

Reply:
The alternative to this is to call `!isBuilderIdConnection()` which is a function a few lines below this. But internally it is calling this exact line: `isSsoConnection(conn, 'builderId')`.

Would using `isBuilderIdConnection()` instead be more clear in your opinion?